### ISSUE TITLE
Styled-components template strings formatting

### ISSUE THREAD

**Issue by RIP21** [prettier/prettier#1872]
I'm not sure if it's can be fixable and how. But it's very common when using `styled-components` to write constructions that look like that.
```javascript
const Something = styled.span`
 ${props => props.right && css`
    padding: 10px 134px 10px 20px;
    &:after {
      left: initial;
      right: 30px;
    }
  `
`
```
So it looks a bit cleaner, especially for multiple rules. After the latest update, it's broken and looks ugly.
```javascript
const Something = styled.span`
  ${props =>
  props.right &&
  css`
    padding: 10px 134px 10px 20px;
    &:after {
      left: initial;
      right: 30px;
    }
  `
`
```
To prove my words, this construction is on the main page first example of styled-components website: https://www.styled-components.com/

**Comment by vjeux** [comment]
Yeah, this is not ideal, I think we can figure out a way to make it better.

**Comment by pomber** [comment]
Would it be OK if prettier assumes that every nested template literal inside a "CSS" template literal also contains CSS?

**Comment by SilencerWeb** [comment]
Are you planning to fix this issue? :)

**Comment by RIP21** [comment]
Yes, I will double that question.
This is really ugly looking in the code :(

**Comment by suchipi** [comment]
PR welcome

**Comment by chuson1996** [comment]
This is still the case for me. Is there any update on this?

**Comment by hegelstad** [comment]
I have the same issue as of 14. nov, so its not only you!

**Comment by evilebottnawi** [comment]
It is easy fix, but what we should do in this case:
```js
const Something = styled.span`
  ${props => props.foo && props.bar && props.baz && props.foobarbaz && props.bazbarfoo && props.somethingOther && css`
      padding: 10px 134px 10px 20px;
      &:after {
        left: initial;
        right: 30px;
      }
  `};
`;
```

**Comment by j-f1** [comment]
@evilebottnawi How about only breaking when it goes past `printWidth`.

**Comment by evilebottnawi** [comment]
@j-f1 yep, just want get feedback

**Comment by vjeux** [comment]
@evilebottnawi, @j-f1 solution seems reasonable to me.

**Comment by Oupsla** [comment]
Hello folks and @evilebottnawi 👋 
You said that it is an easy fix to not add a line break if the line doens't go past `printWidth` (for styled components, or simply css in js).
Do you have any update on this ?

I'm not familiar with the codebase of prettier but I can explore if you show me the way 👍 

Cheers !

**Comment by evilebottnawi** [comment]
@Oupsla 
https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L828

for 
```js
const Something = styled.span`
  ${props => props.right &&
    css`
      padding: 10px 134px 10px 20px;
      &:after {
        left: initial;
        right: 30px;
      }
    `};
`;
```

https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L5658

For:
```js
const Something = styled.span`
  ${props => props.right && css`
      padding: 10px 134px 10px 20px;
      &:after {
        left: initial;
        right: 30px;
      }
    `};
`;
```

Helper what can detect styled style https://github.com/prettier/prettier/blob/master/src/language-js/embed.js#L413 (need move to utils)

**Comment by sergeevNick** [comment]
@evilebottnawi Hi!
Are you going to fix it? As I understand it, the way to fix this is pretty clear.

@Oupsla Hi!
Have you any progress on this?

**Comment by Oupsla** [comment]
Hello @sergeevNick, unfortunately no :confused:

**Comment by Koen028** [comment]
@Oupsla, @evilebottnawi & @vjeux any news on this issue? It's the only thing keeping from using Prettier. 😞

**Comment by evilebottnawi** [comment]
@Koen028 PR welcome, it is open source

**Comment by finnmerlett** [comment]
> @Oupsla Hi!
> 
> https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L828
> 
> For:
> 
> ```js
> const Something = styled.span`
>   ${props => props.right &&
>     css`
>       padding: 10px 134px 10px 20px;
>       &:after {
>         left: initial;
>         right: 30px;
>       }
>     `};
> `;
> ```
> 
> https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L5658
> 
> For:
> 
> ```js
> const Something = styled.span`
>   ${props => props.right && css`
>       padding: 10px 134px 10px 20px;
>       &:after {
>         left: initial;
>         right: 30px;
>       }
>     `};
> `;
> ```
> 
> Helper what detect styled style https://github.com/prettier/prettier/blob/master/src/language-js/embed.js#L413 (need move to utils)
> 
> Maybe something missing.

I know this is super old but this looks like it was pretty close to completion, and wouldn't need very much work at all to add in the needed code. The current formatting for this is really bugging me, it looks incredibly ugly, especially if the expression before the css is very short (as it always adds newlines, regardless). Is there any possibility of an update at all?

**Comment by finnmerlett** [comment]
I would personally chose the  also feel like if the css is short, and the line-width hasn't been hit, that it should allow it to be all on a single line, in a similar way to objects in the form `const obj = { foo: 'bar' };`

**Comment by computnik** [comment]
any updates on this?
